packages/plugable/src/plugable.ts

```diff
@@ -0,0 +1,96 @@
+import { type Plugable, type Key, internals, PlugableInternals, Val } from './types';
+
+const proto = {
+  get<Value>(this: Plugable, key: Key<Value>): Value | undefined {
+    return get(this, key);
+  },
+  set<Value>(
+    this: Plugable,
+    key: Key<Value>,
+    value: Value,
+    isEqual?: (previous: Value | undefined, value: Value) => boolean
+  ) {
+    set(this, key, value, isEqual);
+  },
+  on<K extends Key>(this: Plugable, key: K, listener: (value: Val<K>) => void) {
+    return on(this, key, listener);
+  },
+};
+
+export function createPlugable(): Plugable {
+  const rec = Object.create(proto) as Plugable;
+  rec[internals] = {
+    parent: undefined,
+    listeners: new Map(),
+    handlerToRec: new WeakMap(),
+  };
+  return rec;
+}
+
+export function inheritPlugable(rec: Plugable) {
+  const inherited = Object.create(rec) as Plugable;
+  inherited[internals] = Object.create(rec[internals]) as PlugableInternals;
+  inherited[internals].parent = rec;
+  return inherited;
+}
+
+export function createKey<V = never>(debugName?: string): Key<V> {
+  return Symbol(debugName) as Key<V>;
+}
+
+export function getThrow<Value>(rec: Plugable, key: Key<Value>): Value {
```

Comment:
is `getThrow` used?

Reply:
should be added to the API. and also part of the none prototype API. that I assume I want.